UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs

```diff
@@ -493,17 +493,21 @@ public void CalculateOverallHealth()
 				bool hasAllHeartAttack = true;
 				foreach (var Implant in ImplantList)
 				{
-					var heart = Implant as Heart;
-					if (heart != null)
+					foreach (var bodyPartModification in Implant.BodyPartModifications)
 					{
-						if (heart.HeartAttack == false)
+						var heart = bodyPartModification as Heart;
+						if (heart != null)
 						{
-							hasAllHeartAttack = false;
-							if (ConsciousState != ConsciousState.UNCONSCIOUS)
+							if (heart.HeartAttack == false)
 							{
-								ConsciousState = ConsciousState.UNCONSCIOUS;
+								hasAllHeartAttack = false;
+								if (ConsciousState != ConsciousState.UNCONSCIOUS)
+								{
+									ConsciousState = ConsciousState.UNCONSCIOUS;
```

Comment:
oh boy too many levels of nesting!

Reply:
is this good? xD

Reply:
try exctracting some logic into private methods. This method has a lot of complexity.

Reply:
cleaned up